async-nats/tests/client_tests.rs

```diff
@@ -618,6 +618,42 @@ mod client {
         rx.recv().await;
     }
 
+    #[tokio::test]
+    async fn reconnect_delay_callback() {
+        let server = nats_server::run_basic_server();
+
+        let (tx, mut rx) = tokio::sync::mpsc::unbounded_channel();
+
+        let _ = ConnectOptions::new()
+            .retry_on_initial_connect()
+            .reconnect_delay_callback(move |attempts| {
+                let tx = tx.clone();
+                // println!("reconnect attempts: {attempts}");
+
+                // report back the number of attempts
+                tx.send(attempts).unwrap();
+                // Duration::from_millis(std::cmp::min((attempts * 1000) as u64, 8000))
+                std::time::Duration::from_millis(100)
+            })
+            .connect(server.client_url())
+            .await
+            .unwrap();
+
+        drop(server);
+
+        let attempt = rx.recv().await.unwrap();
```

Comment:
Would be good to check times for default and times for custom. Not just attempts number.